src/ipfsd-daemon.js

```diff
@@ -190,42 +165,29 @@ class Daemon {
     }
 
     let output = ''
+    return new Promise(async (resolve, reject) => {
+      this.subprocess = execa(this.exec, args, {
+        env: this.env
+      })
+      this.subprocess.stderr.on('data', data => daemonLog.err(data.toString()))
+      this.subprocess.stdout.on('data', data => {
+        daemonLog.info(data.toString())
+        output += data
+        const apiMatch = output.trim().match(/API .*listening on:? (.*)/)
+        const gwMatch = output.trim().match(/Gateway .*listening on:? (.*)/)
+
+        if (apiMatch && apiMatch.length > 0) {
+          this.setApi(apiMatch[1])
+        }
 
-    return new Promise(async (resolve, reject) => { // eslint-disable-line no-async-promise-executor
-      this.subprocess = run(this, args, {
-        env: this.env,
-        stderr: (data) => {
-          data = String(data)
-
-          if (data) {
-            daemonLog.err(data.trim())
-          }
-        },
-        stdout: (data) => {
-          data = String(data)
-
-          if (data) {
-            daemonLog.info(data.trim())
-          }
-
-          output += data
-
-          const apiMatch = output.trim().match(/API .*listening on:? (.*)/)
-          const gwMatch = output.trim().match(/Gateway .*listening on:? (.*)/)
-
-          if (apiMatch && apiMatch.length > 0) {
-            this.setApi(apiMatch[1])
-          }
-
-          if (gwMatch && gwMatch.length > 0) {
-            this.setGateway(gwMatch[1])
-          }
-
-          if (output.match(/(?:daemon is running|Daemon is ready)/)) {
-            // we're good
-            this.started = true
-            resolve(this.api)
-          }
+        if (gwMatch && gwMatch.length > 0) {
+          this.setGateway(gwMatch[1])
+        }
+
+        if (output.match(/(?:daemon is running|Daemon is ready)/)) {
+          // we're good
+          this.started = true
+          resolve(this.api)
```

Comment:
This will continue appending any logs to `output` after resolving the promise which causes a memory leak.

We should have two listeners for `stdout.on('data')` - one that calls `daemonLog.info(data.toString())` and the other that waits for the `Daemon is ready` type messages and unregisters itself when it before resolving the promise.

Reply:
Doesn't need to be fixed here though :)